### DISCUSSION TITLE
Async routes

### DISCUSSION THREAD

**Discussion by transbitch** [solidjs/solid-start#1468] [Q&A]
In SSR, it would appear to me that there is no easy way to asynchronously load data before rendering the component. Docs suggest using `"use server"`, but that's not what I want -- I do not want to generate an RPC route.

`createHandler` is async, so you could add data fetching there. But then you have to do your routing there, before `App` is rendered. And it's not easy to pass in your data to `App` either because of the implicit control flow from `StartServer` to `App`.

Edit: I did try `createResource`. The server sends the fallback, it doesn't wait for the resource.

**Comment by timredd** [comment]
Take a look at [createAsync](https://docs.solidjs.com/solid-router/reference/data-apis/create-async#createasync) - sounds like it may be what you're looking for?

**Reply by aquaductape** [reply]
Make sure to enable deferStream with createAsync `createAsync(() => getUser(), { deferStream: true })`

**Reply by transbitch** [reply]
I'm not using solid router, so I don't have access to this primitive.

**Comment by devagrawal09** [comment]
If I'm understanding you correctly, you want to block the rendering of the component until the data has finished fetching, and only start rendering the component once it's finished loading?
Or do you just want to not display anything on the screen until the data has finished loading?

The second thing can be done by wrapping the JSX in a `<Suspense` boundary without any fallback.

As for the first thing, it's generally not a good idea to block the rendering of a component at all. Solid can continue to render the component while the data is fetching, so that once the fetching is complete it has to do minimal work in updating the exact nodes that depend on the async data.

**Reply by transbitch** [reply]
I would like to serve static HTML from the server. Is there a way that I can do this with `Suspense`?

**Comment by doeixd** [comment]
Try passing `"async"` to the `ssr` property in the `defineConfig` object.

```js
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  ssr: "async"
  // The rest of your config here....
})
```
